packages/react/react/src/hooks/useVisibilityEvent.spec.tsx

```diff
@@ -0,0 +1,41 @@
+import { fireEvent, render, renderHook } from '@testing-library/react';
+import { useState } from 'react';
+import { useVisibilityEvent } from './useVisibilityEvent';
+
+type VisibilityState = Document['visibilityState'];
+
+describe('useVisibilityEvent', () => {
+  it('When a user leaves the page.', () => {
+    const callbackMock = jest.fn();
+    renderHook(() => useVisibilityEvent(callbackMock));
+    expect(document.visibilityState).toBe('visible');
+
+    Object.defineProperty(document, 'visibilityState', { value: 'hidden' });
+    fireEvent(document, new Event('visibilitychange'));
+
+    expect(document.visibilityState).toBe('hidden');
```

Comment:
What about writing the test code as below? I think it adds an abstraction layer that makes the code easier to read.

```tsx
const handleVisibilityChange = jest.fn();
renderHook(() => useVisibilityEvent(handleVisibilityChange));

mockVisibilityState('visible');

expect(handleVisibilityChange).toBeCalledWith('visible');
```

```ts
function mockVisibilityState(visibilityState: Document['visibilityState']) {
  Object.defineProperty(document, 'visibilityState', { value: visibilityState });
  fireEvent(document, new Event('visibilitychange'));
}
```